test/Dialect/FIRRTL/lowerXMR.mlir

```diff
@@ -692,3 +692,55 @@ firrtl.circuit "ForceRelease" {
       firrtl.ref.release_initial %c, %r_p : !firrtl.uint<1>, !firrtl.rwprobe<uint<4>>
     }
   }
+
+// -----
+// Check tracking of public output refs as sv.macro.decl and sv.macro.def
+
+// CHECK-LABEL: sv.macro.decl @ref_Top_Top_a
+// CHECK-NEXT:  sv.macro.decl @ref_Top_Top_b
+// CHECK-NEXT:  sv.macro.decl @ref_Top_Top_c
+// CHECK-NEXT:  sv.macro.decl @ref_Top_Top_d
+// CHECK-NOT:   sv.macro.decl @ref_Top_Top_e
+// CHECK-NEXT:  sv.macro.decl @ref_Top_Foo_x
+// CHECK-NEXT:  sv.macro.decl @ref_Top_Foo_y
+
+// CHECK-LABEL: firrtl.circuit "Top"
+firrtl.circuit "Top" {
+  // CHECK-LABEL: hw.hierpath private @xmrPath [@Top::@foo, @Foo::@x]
+  // CHECK-NEXT:  sv.macro.def @ref_Top_Top_a "{{[{][{]0[}][}]}}"([#hw.innerNameRef<@Top::@w>]) {output_file = #hw.output_file<"ref_Top_Top.sv">}
+  // CHECK-NEXT:  sv.macro.def @ref_Top_Top_b "{{[{][{]0[}][}]}}"([@xmrPath]) {output_file = #hw.output_file<"ref_Top_Top.sv">}
+  // CHECK-NEXT:  sv.macro.def @ref_Top_Top_c "{{[{][{]0[}][}]}}.internal.path"([#hw.innerNameRef<@Top::@foo>]) {output_file = #hw.output_file<"ref_Top_Top.sv">}
+  // CHECK-NEXT:  sv.macro.def @ref_Top_Top_d "{{[{][{]0[}][}]}}"([#hw.innerNameRef<@Top::@xmr_sym>]) {output_file = #hw.output_file<"ref_Top_Top.sv">}
+  // CHECK-NEXT:  sv.macro.def @ref_Top_Foo_x "{{[{][{]0[}][}]}}"([#hw.innerNameRef<@Foo::@x>]) {output_file = #hw.output_file<"ref_Top_Foo.sv">}
+  // CHECK-NEXT:  sv.macro.def @ref_Top_Foo_y "internal.path" {output_file = #hw.output_file<"ref_Top_Foo.sv">}
```

Comment:
Suggestion: `CHECK-NEXT-LITERAL` would let you match `{{0}}` directly without the regex escaping.

Informational: I find long matches hard to parse and will often split them up where the `output_file` is a `CHECK-SAME` on the next line with indentation. Only informational.

Reply:
I'm very happy to learn about `CHECK-NEXT-LITERAL`, `{{[{][{]0[}][}]}}` looks like a nightmare

Reply:
Updated to be more readable.
FYI it's `CHECK-NEXT{LITERAL}`.